clippy_lints/src/utils/internal_lints.rs

```diff
@@ -395,3 +401,111 @@ fn is_trigger_fn(fn_kind: FnKind<'_>) -> bool {
         FnKind::Closure(..) => false,
     }
 }
+
+declare_lint_pass!(CollapsibleCalls => [COLLAPSIBLE_SPAN_LINT_CALLS]);
+
+impl  EarlyLintPass for CollapsibleCalls {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &ast::Expr) {
+        use ast::ExprKind;
+
+        if_chain! {
+            if let ExprKind::Call(ref func, ref and_then_args) = expr.kind;
+            if let ExprKind::Path(None, ref path) = func.kind;
+            if let match_path_ast(path, &["span_lint_and_then"]);
+            if and_then_args.len() == 5;
+            if let ExprKind::Closure(_, _, _, _, block) = and_then_args[4].kind;
+            if let ExprKind::Block(block, _) = block.kind;
```

Comment:
Rather use `utils::remove_blocks` here.

Reply:
Wait `remove_blocks` doesn't resolve `{ x; }`->`x;` So we need to check for all three cases:

```rust
|db| { db._ }

|db| { db._; }

|db| db._
```

Reply:
```rust
|db| { db._ }
|db| db._
```
The last two forms emits error.

Reply:
`|db| { db._; }` should also produce an error.